render/mapping.go

```diff
@@ -365,26 +220,51 @@ func MapEndpoint2Pseudo(n RenderableNode, local report.Networks) RenderableNodes
 // format for a process, but without any Major or Minor labels.
 // It does not have enough info to do that, and the resulting graph
 // must be merged with a process graph to get that info.
-func MapEndpoint2Process(n RenderableNode, local report.Networks) RenderableNodes {
+func MapEndpoint2Process(n report.Node, local report.Networks) report.Nodes {
 	// Nodes without a hostid are treated as pseudo nodes
 	if _, ok := n.Latest.Lookup(report.HostNodeID); !ok {
 		return MapEndpoint2Pseudo(n, local)
 	}
 
-	pid, ok := n.Node.Latest.Lookup(process.PID)
+	pid, ok := n.Latest.LookupEntry(process.PID)
 	if !ok {
-		return RenderableNodes{}
+		return report.Nodes{}
 	}
 
-	id := MakeProcessID(report.ExtractHostID(n.Node), pid)
-	node := NewDerivedNode(id, n.WithParents(report.EmptySets))
-	node.Shape = Square
-	node.Children = node.Children.Add(n)
-	return RenderableNodes{id: node}
+	id := report.MakeProcessNodeID(report.ExtractHostID(n), pid.Value)
+	node := NewDerivedNode(id, n).WithTopology(report.Process)
+	node.Latest = node.Latest.Set(process.PID, pid.Timestamp, pid.Value)
+	node.Counters = node.Counters.Add(n.Topology, 1)
+	return report.Nodes{id: node}
+}
+
+// MapTo re-indexes a set of nodes by some key from Latest, changing the topology
+// TODO(paulbellamy): not sure this works, or is general enough
+func MapTo(topology, key string) func(report.Node, report.Networks) report.Nodes {
```

Comment:
Yeah. had put it in as an experiment in generalizing the MapX2Y functions. Was curious if you think it's worth pursuing, or bailing on that.